### ISSUE TITLE
React integration

### ISSUE THREAD

**Issue by benoj** [artf/grapesjs#205]
Hey – love the project!

I have an existing project which I am trying to integrate this into, however this is a react application. At present I am loading the editor in an Iframe (is there a better solution?). Is there any support for React?

Thanks,

Ben

**Comment by artf** [comment]
Actually, there is no particular support but I think you just have to create a wrapper and call the init method inside componentDidMount

**Comment by jbmusso** [comment]
The following in the React documentation should be useful: https://reactjs.org/docs/integrating-with-other-libraries.html

**Comment by aditya-dikonia** [comment]
@jbmusso I checked the given link but don't understand how to integrate this component with the reactjs. Can you please explain it in brief?

**Comment by jbmusso** [comment]
Sure. In a typical React application, React will handle all DOM manipulations for you up from the root component (say, `<App/>`) down to the leaves. In the present case, you want a way to tell React to stop interacting with a specific portion of the DOM, so that part should be handled by Grape.js instead.

The React documentation suggests that you implement a `React.Component` with a super simple render method:

```javascript
render() {
    return <div ref={el => this.el = el} />;
  }
```
The important part is `ref={el => this.el = el}` so you can keep track, in that component, of the actual DOM node which the third party library should be interacting with (Grape, in the present case). When the component is mounted into the DOM, you can then interact with that `ref` and call Grape methods to do whatever it should be doing with the DOM under that specific portion of your app.
`componentWillUnmount` is especially useful for cleaning stuffs started by the 3rd party library, such as `setInterval` or event listeners, in order to avoid some memory leaks.

Lifecycle:
```
-> mount once / prepare (componentDidMount)
-> render one or more times (render)
-> cleanup once / clean-up stuffs (componentWillUnmount)
```

Try to imagine the DOM as a canvas and React as a brush. In the present case, you're just adding one more brush (Grape) that will draw/paint stuffs on the same canvas (note that I'm not talking about HTML5 canvas here). The developer is the artist.

tldr; You just need to implement a component that tells React that it should stop painting on that specific part of the DOM, but delegate that responsibility to another library.

**Comment by artf** [comment]
Thanks @jbmusso for such detailed explanation
